File: ios/Freepass/Freepass/Entry.swift
import SwiftCBOR

class Entry {
	var fields: [(String, Field)] = []

	init?(fromCbor arr: CBOR) {
		guard case let CBOR.Map(field_objs)? = arr[0]?["fields"] else { return nil }
		field_objs.forEach {
			guard case let CBOR.UTF8String(key) = $0 else { return }
			guard let value = Field(fromCbor: $1) else { return }
			fields.append((key, value))
		}
	}

	init() {}
}

File: android/app/src/main/java/technology/unrelenting/freepass/Entry.kt
package technology.unrelenting.freepass

import com.upokecenter.cbor.CBORObject

data class Entry(var fields: Map<String, Field> = emptyMap()) {
	companion object {
        fun fromCbor(cborObject: CBORObject): Entry {
			val result = mutableMapOf<String, Field>()
			val fields = cborObject["fields"]
			fields.keys.forEach {
				result.put(it.AsString(), Field.fromCbor(fields[it]))
			}
			return Entry(result)
        }
	}
}